encoding/gzip/gzip.go

```diff
@@ -46,6 +47,23 @@ type writer struct {
 	pool *sync.Pool
 }
 
+// SetLevel updates the registered gzip compressor to use the compression level specified (gzip.HuffmanOnly is not supported).
+// NOTE: this function must only be called during initialization time (i.e. in an init() function),
+// and is not thread-safe.
+//
+// The error returned will be nil if the specified level is valid.
+func SetLevel(level int) error {
+	if level < gzip.DefaultCompression || level > gzip.BestCompression {
+		return fmt.Errorf("grpc: invalid gzip compression level: %d", level)
+	}
+	c := encoding.GetCompressor(Name).(*compressor)
+	c.poolCompressor.New = func() interface{} {
+		w, _ := gzip.NewWriterLevel(ioutil.Discard, level)
```

Comment:
This should never fail, but let's not just blindly ignore `err`.  Check it and `panic`, please (since there is no capability to return an error, and we expect that it is impossible).  Same in `rpc_util.go`.